Typescript compilation error when `verbatimModuleSyntax` is enabled on generated interface.ts

**Reproduction**
Include the code that is causing the error:

schema:
```
❯ cat dbschema/default.esdl
module default {
  type Image {
    required path: str;
  }
}
```

`pnpx @edgedb/generate interfaces`, enable `verbatimModuleSyntax` in tsconfig.json, and run tsc with the generated `dbschema/interface.ts`

**Describe the bug**
tsc complains with:
```
dbschema/interfaces.ts:71:1 - error TS1288: An import alias cannot resolve to a type or type-only declaration when 'verbatimModuleSyntax' is enabled.

71 import Image = $default.Image;
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error in dbschema/interfaces.ts:71
```
 
**Expected behavior**
A clear and concise description of what you expected to happen. If applicable, add screenshots to help explain your problem.

Typescript compile with no errors.

**Versions (please complete the following information):**

<!--
For EdgeDB Version: run `edgedb query 'select sys::get_version_as_str()'` from your project directory (or run `select sys::get_version_as_str();` in the EdgeDB interactive shell).
For EdgeDB CLI version: Run `edgedb --version` from anywhere
For `edgedb-js` version: Run `npm list edgedb --depth=0` from your project directory
For Node/Deno version: Run `node --version` or `deno --version`
-->

- OS: Ubuntu 22.04.1 LTS
- EdgeDB CLI version (e.g. `2.0`): 4.0.2
- `edgedb-js` version (e.g. `0.20.10`;): 1.5.4

**Additional context**
The code that generates that line is added in https://github.com/edgedb/edgedb-js/commit/6c7de1a0b684dd0e734eff8b1efd1f94baa07103

--- a/packages/generate/src/edgeql-js/generateInterfaces.ts
+++ b/packages/generate/src/edgeql-js/generateInterfaces.ts
@@ -203,22 +203,11 @@ export const generateInterfaces = (params: GenerateInterfacesParams) => {
           (nestedMod) => nestedMod.fullInternalName
         ),
       ];
+      const sliceTo = module.internalName.length + 1;
       for (const typeRef of typeRefs) {
-        plainTypesCode.writeln([
-          `import ${typeRef.slice(
-            module.internalName.length + 1
-          )} = ${typeRef};`,
-        ]);
+        const aliased = typeRef.slice(sliceTo);
+        plainTypesCode.writeln([`export type ${aliased} = ${typeRef};\n`]);
       }
-      plainTypesCode.writeln([t`export type {`]);
-      plainTypesCode.increaseIndent();
-      plainTypesCode.writeln([
-        typeRefs
-          .map((typeRef) => typeRef.slice(module.internalName.length + 1))
-          .join(",\n"),
-      ]);
-      plainTypesCode.decreaseIndent();
-      plainTypesCode.writeln([t`};`]);
     }
   };
 

--- a/packages/generate/src/edgeql-js/generateInterfaces.ts
+++ b/packages/generate/src/edgeql-js/generateInterfaces.ts
@@ -197,16 +197,25 @@ export const generateInterfaces = (params: GenerateInterfacesParams) => {
     plainTypesCode.addExport(module.internalName, { modes: ["js"] });
 
     if (module.isRoot && module.name === "default") {
-      const typeRefs = [
-        ...module.types.values(),
-        ...[...module.nestedModules.values()].map(
-          (nestedMod) => nestedMod.fullInternalName
-        ),
-      ];
       const sliceTo = module.internalName.length + 1;
-      for (const typeRef of typeRefs) {
+      for (const typeRef of module.types.values()) {
         const aliased = typeRef.slice(sliceTo);
         plainTypesCode.writeln([`export type ${aliased} = ${typeRef};\n`]);
+      }
+
+      for (const nestedMod of module.nestedModules.values()) {
+        plainTypesCode.writeln([
+          `export namespace ${nestedMod.internalName} {`,
+        ]);
+        plainTypesCode.increaseIndent();
+        for (const typeRef of nestedMod.types.values()) {
+          const aliased = typeRef.slice(
+            sliceTo + nestedMod.internalName.length + 1
+          );
+          plainTypesCode.writeln([`export type ${aliased} = ${typeRef};`]);
+        }
+        plainTypesCode.decreaseIndent();
+        plainTypesCode.writeln([`}`]);
       }
     }
   };

